### PR TITLE
add compatibility with docker on mac

### PR DESCRIPTION
**Pull request by id**
i.e. make it possible to test brod as is on both mac and linux

### CODE REVIEW

**Comment by zmstone** [comment]
the hostname for zookeeper is from `--link` option in docker run command for kafka container.
how does 'localhost' work ?

**Comment by zmstone** [comment]
ok, I see the pausenet magic in your PR for brod.
perhaps parameterise the zookeeper hostname instead of hard-code it to 'localhost'.
also, please update README

**Comment by id** [comment]
Updated README, added docker-compose instead of docker commands.
As for zookeeper hostname, is it really important? It was hardcoded to `zookeeper` before.

**Comment by id** [comment]
Tested on OSX and Linux.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -27,15 +27,16 @@ fi
 ipwithnetmask="$(ip -f inet addr show dev eth0 | awk '/inet / { print $2 }')"
 ipaddress="${ipwithnetmask%/*}"
 
+[ -z "$ADVERTISED_HOSTNAME" ] && ADVERTISED_HOSTNAME="localhost"
```
Comment (zmstone) [inline docker-entrypoint.sh]: please default it to `ipaddress` so it's backward compatible.
(and you have this set explicitly in the compose file, which is good)

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -55,15 +56,15 @@ fi
 
 wait_for_kafka() {
   echo '### waiting for kafka to be ready'
-  if ! kafka-topics.sh --zookeeper zookeeper --list >/dev/null 2>&1; then
+  if ! kafka-topics.sh --zookeeper localhost --list >/dev/null 2>&1; then
```
Comment (zmstone) [inline docker-entrypoint.sh]: and parameterise zookeeper host too in docker's env.

File: server.properties [hunk 3]
```diff
@@ -114,7 +114,7 @@ log.retention.check.interval.ms=300000
 # You can also append an optional chroot string to the urls to specify the
 # root directory for all kafka znodes.
 ### changed for brod
-zookeeper.connect=zookeeper:2181
+zookeeper.connect=localhost:2181
```
Comment (zmstone) [inline server.properties:117]: then a `sed` is needed to replace it with the parameterised zookeeper hostname.

**Pull request merged**